tests/common.go

```diff
@@ -1143,27 +1144,31 @@ func ValidatePureVolumeStatisticsDynamicUpdate(ctx *scheduler.Context, errChan .
 			vols, err = Inst().S.GetVolumes(ctx)
 			processError(err, errChan...)
 		})
-		byteUsedInitial, err := Inst().V.ValidateGetByteUsedForVolume(vols[0].ID, make(map[string]string))
-		fmt.Printf("initially the byteUsed is %v\n", byteUsedInitial)
-		// get the pod for this pvc
-		pods, err := Inst().S.GetPodsForPVC(vols[0].Name, vols[0].Namespace)
-		processError(err, errChan...)
+		// skiping ValidatePureVolumeStatisticsDynamicUpdate test for raw block volumes. Need to change getStats method
+		if !vols[0].Raw {
+			byteUsedInitial, err := Inst().V.ValidateGetByteUsedForVolume(vols[0].ID, make(map[string]string))
+			fmt.Printf("initially the byteUsed is %v\n", byteUsedInitial)
 
-		mountPath, bytesToWrite := pureutils.GetAppDataDir(pods[0].Namespace)
+			// get the pod for this pvc
+			pods, err := Inst().S.GetPodsForPVC(vols[0].Name, vols[0].Namespace)
+			processError(err, errChan...)
 
-		// write to the Direct Access volume
-		ddCmd := fmt.Sprintf("dd bs=512 count=%d if=/dev/urandom of=%s/myfile", bytesToWrite/512, mountPath)
-		cmdArgs := []string{"exec", "-it", pods[0].Name, "-n", pods[0].Namespace, "--", "bash", "-c", ddCmd}
-		err = osutils.Kubectl(cmdArgs)
-		processError(err, errChan...)
-		fmt.Println("sleeping to let volume usage get reflected")
-		// wait until the backends size is reflected before making the REST call
-		time.Sleep(time.Minute * 2)
+			mountPath, bytesToWrite := pureutils.GetAppDataDir(pods[0].Namespace)
+			mountPath = mountPath + "/myfile"
 
-		byteUsedAfter, err := Inst().V.ValidateGetByteUsedForVolume(vols[0].ID, make(map[string]string))
-		fmt.Printf("after writing random bytes to the file the byteUsed in volume %s is %v\n", vols[0].ID, byteUsedAfter)
-		expect(byteUsedAfter > byteUsedInitial).To(beTrue(), "bytes used did not increase after writing random bytes to the file")
+			// write to the Direct Access volume
+			ddCmd := fmt.Sprintf("dd bs=512 count=%d if=/dev/urandom of=%s", bytesToWrite/512, mountPath)
+			cmdArgs := []string{"exec", "-it", pods[0].Name, "-n", pods[0].Namespace, "--", "bash", "-c", ddCmd}
+			err = osutils.Kubectl(cmdArgs)
+			processError(err, errChan...)
+			fmt.Println("sleeping to let volume usage get reflected")
+			// wait until the backends size is reflected before making the REST call
+			time.Sleep(time.Minute * 2)
```

Comment:
Is there any other way to make sure size is reflecting instead of sleeping for 2 minutes ?

Reply:
We will keep it as it is as we have that standard time for FA or FB volumes. we will discuss more on this and reflect the changes after that.